src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,861 @@
+        Class<?> result = null;
+
+        if (field != null) {
+            result = field.getType();
+        }
+        if (result == List.class || result == Set.class) {
+            final ParameterizedType type = (ParameterizedType) field.getGenericType();
+            final Class<?> parameterClass = (Class<?>) type.getActualTypeArguments()[0];
+
+            if (parameterClass == Integer.class) {
+                result = int[].class;
+            }
+            else if (parameterClass == String.class) {
+                result = String[].class;
+            }
+            else if (parameterClass == Pattern.class) {
+                result = Pattern[].class;
+            }
+            else {
+                final String message = "Unknown parameterized type: "
+                        + parameterClass.getSimpleName();
+                throw new MacroExecutionException(message);
+            }
+        }
+        else if (result == BitSet.class) {
+            result = int[].class;
+        }
+
+        return result;
+    }
+
+    /**
+     * Gets the field with the given name from the given class.
+     *
+     * @param fieldClass the class to get the field from.
+     * @param propertyName the name of the field.
+     * @return the field we are looking for.
+     */
+    private static Field getField(Class<?> fieldClass, String propertyName) {
+        Field result = null;
+        Class<?> currentClass = fieldClass;
+
+        while (!Object.class.equals(currentClass)) {
+            try {
+                result = currentClass.getDeclaredField(propertyName);
+                result.trySetAccessible();
+                break;
+            }
+            catch (NoSuchFieldException ignored) {
+                currentClass = currentClass.getSuperclass();
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Get a set of properties for the given class.
+     *
+     * @param clss the class to get the properties for.
+     * @return a set of properties for the given class.
+     */
+    private static Set<String> getProperties(Class<?> clss) {
+        final Set<String> result = new TreeSet<>();
+        final PropertyDescriptor[] map = PropertyUtils.getPropertyDescriptors(clss);
+
+        for (PropertyDescriptor propertyDescriptor : map) {
+            if (propertyDescriptor.getWriteMethod() != null) {
+                result.add(propertyDescriptor.getName());
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Fixes the captured properties. This method removes properties that are not documented and
+     * adds properties that have to be documented but are not captured.
+     *
+     * @param instance the instance of the module.
+     * @param clss the class of the module.
+     * @param properties the properties of the module.
+     */
+    private static void fixCapturedProperties(
+            Object instance, Class<?> clss, Set<String> properties) {
```

Comment:
can we split this method to
`removeNotProperties`
`addNonExplicitProperties` ?

it will be easier to understand, as for now it is "fix" ==== "do something unclear"

Reply:
Simplified.